sound/soc/sof/pm.c

```diff
@@ -205,8 +212,16 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 	target_state = snd_sof_dsp_power_target(sdev);
 
 	/* Skip to platform-specific suspend if DSP is entering D0 */
-	if (target_state == SOF_DSP_PM_D0)
+	if (target_state == SOF_DSP_PM_D0) {
+		ret = snd_sof_dma_trace_trigger(sdev, SNDRV_PCM_TRIGGER_STOP);
+		if (ret < 0)
+			dev_err(sdev->dev,
+				"error: snd_sof_dma_trace_trigger: stop: %d\n", ret);
+		else
+			sdev->dtrace_is_enabled = false;
```

Comment:
I think L216-221 code would be better moved to trace.c and manage all logic around "sdev->dtrace_is_enabled" in trace.c and just have simple function calls here.

Reply:
Sure, I will do it.

Reply:
@kv2019i @ranj063 Based on your suggestion, what about put the check code and error handling code in the definition of snd_sof_dma_trace_trigger() in ops.h? With this, we can make the code platform independent and don't create a new function.

Reply:
@libinyang Sounds good to me. Maybe spin a version for review?

Reply:
> @libinyang Sounds good to me. Maybe spin a version for review?

OK. Thanks.